itou/siaes/migrations/0029_auto_20200821_1616.py

```diff
@@ -0,0 +1,42 @@
+# Generated by Django 3.1 on 2020-08-21 14:16
+
+import datetime
+
+from django.db import migrations
+
+from itou.siaes.models import Siae
+
+
+def fix_siae_source(apps, schema_editor):
+    for siae in Siae.objects.filter(source=Siae.SOURCE_USER_CREATED):
+        user = siae.created_by
+        if user:
+            assert not user.is_prescriber
+            assert not user.is_superuser
+            if user.is_siae_staff:
+                # Siae created by a regular siae user - do not touch.
+                assert not user.is_job_seeker
+                assert not user.is_staff
+            else:
+                # Siae created by our staff. Should be STAFF_CREATED.
+                assert user.is_staff
+                # Some staff users are job seekers ¯\_(ツ)_/¯
```

Comment:
> Some staff users are job seekers

Ça c'est parce que je demande aux admins de se créer un compte sur Itou en prenant le tunnel le plus simple : celui des DE. Ça n'est pas grave en soi.

Reply:
Du coup tu n'as pas besoin de tous ces `assert` en fait. La seule condition importante c'est `siae.created_by.is_staff`.

Imagine si toi tu avais créé une SIAE dans l'admin pour dépanner le support : tu serais à la fois `is_superuser` et `is_staff`. Du coup ta migration ne passerait pas et prendrait fin à `assert not user.is_superuser`.

Reply:
Effectivement j'ai l'habitude d'utiliser un max d'assertion dans tous mes scripts pour contrôler au vol la qualité et la cohérence des données que je manipule. Dans un script django admin cela ne dérange personne, en revanche dans une migration c'est chaud car cela peut casser la prod en ratant la MEP!

Du coup je vais virer un max d'assertions de la migration vu que c'est seulement du one-shot et que j'ai la conscience tranquille vu que j'ai tout testé sur le dernier snapshot de production.